### PR TITLE
[app_publisher] Enable multiple condition

### PR DESCRIPTION
**Pull request by 708yamaguchi**
I enabled multiple conditions in app_publisher.

If any one of the cond is true, the topic will be published.

This PR includes https://github.com/knorth55/app_manager_utils/pull/47

### CODE REVIEW

File: app_publisher/src/app_publisher/rostopic_publisher_plugin.py [hunk 1]
```diff
@@ -47,9 +47,15 @@ def _check_condition(self, topic_cond, ctx):
             return False
 
     def _publish_topic(self, topic, ctx, check_cond=False):
-        if (check_cond and 'cond' in topic
-                and self._check_condition(topic['cond'], ctx)):
-            return
+        if check_cond and 'cond' in topic:
+            conditions = topic['cond']
+            if isinstance(conditions, list) is False:
+                conditions = list(conditions)
```
Comment (knorth55) [inline app_publisher/src/app_publisher/rostopic_publisher_plugin.py]:
```suggestion
            if not isinstance(conditions, list):
                conditions = [conditions]
```

File: app_publisher/src/app_publisher/rostopic_publisher_plugin.py [hunk 2]
```diff
@@ -47,9 +47,15 @@ def _check_condition(self, topic_cond, ctx):
...
+            can_publish = False
```
Comment (knorth55) [inline app_publisher/src/app_publisher/rostopic_publisher_plugin.py]:
```suggestion
            do_publish = False
```

**Review by knorth55 — CHANGES_REQUESTED** [review]

**Comment by 708yamaguchi** [comment]
Thank you for your review!
I updated.

By the way, could you please teach me why you prefer
```
if not isinstance(conditions, list):
```
to
```
if isinstance(conditions, list) is False:
```
and prefer
```
conditions = [conditions]
```
to
```
conditions = list(conditions)
```
?

**Comment by knorth55** [comment]
in cpython, `is` is OK for boolean comparison, but not safe for other implementation.
so `is` is no recommended for boolean comparison.

`list()` means type conversion from `dict` or `tuple`.
https://stackoverflow.com/questions/23703109/difference-between-using-and-list-in-python

**Comment by 708yamaguchi** [comment]
I see.
Thank you very much for your explanation!

**Review by knorth55 — APPROVED** [review]

**Pull request merged**